File: clients/java/android/src/main/java/com/wordnik/client/model/Trade.java
package com.wordnik.client.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;
public class Trade {
  @JsonProperty("timestamp")
  private Date timestamp = null;
  @JsonProperty("symbol")
  private String symbol = null;
  @JsonProperty("side")
  private String side = null;
  @JsonProperty("size")
  private Double size = null;
  @JsonProperty("price")
  private Double price = null;
  @JsonProperty("tickDirection")
  private String tickDirection = null;
  @JsonProperty("trdMatchID")
  private String trdMatchID = null;
  @JsonProperty("grossValue")
  private Double grossValue = null;
  @JsonProperty("homeNotional")
  private Double homeNotional = null;
  @JsonProperty("foreignNotional")
  private Double foreignNotional = null;
  @JsonProperty("id")
  private Double id = null;
  public Date getTimestamp() {
    return timestamp;
  }
  public void setTimestamp(Date timestamp) {
    this.timestamp = timestamp;
  }

  public String getSymbol() {
    return symbol;
  }
  public void setSymbol(String symbol) {
    this.symbol = symbol;
  }

  public String getSide() {
    return side;
  }
  public void setSide(String side) {
    this.side = side;
  }

  public Double getSize() {
    return size;
  }
  public void setSize(Double size) {
    this.size = size;
  }

  public Double getPrice() {
    return price;
  }
  public void setPrice(Double price) {
    this.price = price;
  }

  public String getTickDirection() {
    return tickDirection;
  }
  public void setTickDirection(String tickDirection) {
    this.tickDirection = tickDirection;
  }

  public String getTrdMatchID() {
    return trdMatchID;
  }
  public void setTrdMatchID(String trdMatchID) {
    this.trdMatchID = trdMatchID;
  }

  public Double getGrossValue() {
    return grossValue;
  }
  public void setGrossValue(Double grossValue) {
    this.grossValue = grossValue;
  }

  public Double getHomeNotional() {
    return homeNotional;
  }
  public void setHomeNotional(Double homeNotional) {
    this.homeNotional = homeNotional;
  }

  public Double getForeignNotional() {
    return foreignNotional;
  }
  public void setForeignNotional(Double foreignNotional) {
    this.foreignNotional = foreignNotional;
  }

  public Double getId() {
    return id;
  }
  public void setId(Double id) {
    this.id = id;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class Trade {\n");
    sb.append("  timestamp: ").append(timestamp).append("\n");
    sb.append("  symbol: ").append(symbol).append("\n");
    sb.append("  side: ").append(side).append("\n");
    sb.append("  size: ").append(size).append("\n");
    sb.append("  price: ").append(price).append("\n");
    sb.append("  tickDirection: ").append(tickDirection).append("\n");
    sb.append("  trdMatchID: ").append(trdMatchID).append("\n");
    sb.append("  grossValue: ").append(grossValue).append("\n");
    sb.append("  homeNotional: ").append(homeNotional).append("\n");
    sb.append("  foreignNotional: ").append(foreignNotional).append("\n");
    sb.append("  id: ").append(id).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

File: clients/dart/lib/model/trade.dart
part of api;


@Entity()
class Trade {
  
  @Property(name: 'timestamp')
  DateTime timestamp = null;
  

  @Property(name: 'symbol')
  String symbol = null;
  

  @Property(name: 'side')
  String side = null;
  

  @Property(name: 'size')
  Number size = null;
  

  @Property(name: 'price')
  double price = null;
  

  @Property(name: 'tickDirection')
  String tickDirection = null;
  

  @Property(name: 'trdMatchID')
  String trdMatchID = null;
  

  @Property(name: 'grossValue')
  Number grossValue = null;
  

  @Property(name: 'homeNotional')
  double homeNotional = null;
  

  @Property(name: 'foreignNotional')
  double foreignNotional = null;
  

  @Property(name: 'id')
  double id = null;
  
  Trade();

  @override
  String toString()  {
    return 'Trade[timestamp=$timestamp, symbol=$symbol, side=$side, size=$size, price=$price, tickDirection=$tickDirection, trdMatchID=$trdMatchID, grossValue=$grossValue, homeNotional=$homeNotional, foreignNotional=$foreignNotional, id=$id, ]';
  }

}

